reporters/jest/TerraVerboseReporter.js

```diff
@@ -0,0 +1,118 @@
+const Logger = require('../../scripts/utils/logger');
+
+const LOG_CONTEXT = '[Terra-Toolkit:terra-verbose-reporter]';
+
+class TerraVerboseReporter extends VerboseReporter {
+  constructor(globalConfig) {
+    super(globalConfig);
+    this.results = {
+      startDate: '',
+      output: {},
+      endDate: '',
+    };
+    this.unformattedResult = {};
+    this.moduleName = '';
+    this.log = this.log.bind(this);
+    this.ensureResultsDir = this.ensureResultsDir.bind(this);
+    this.setTestModule = this.setTestModule.bind(this);
+    this.setTestDirPath = this.setTestDirPath.bind(this);
+    this.setResultDir = this.setResultDir.bind(this);
+    this.getIsMonoRepo = this.getIsMonoRepo.bind(this);
+    this.isMonoRepo = this.getIsMonoRepo();
+    this.filePath = this.setTestDirPath();
+    this.resultDir = this.setResultDir(globalConfig);
+    this.ensureResultsDir();
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  setTestDirPath() {
+    let testDir = 'tests';
+    if (fs.existsSync(path.join(process.cwd(), '/test'))) {
+      testDir = 'test';
+    }
+    return path.join(testDir, 'jest', 'reports', 'results');
+  }
+
+  setResultDir(globalConfig) {
+    if (!globalConfig.rootDir) {
+      return path.join(process.cwd(), this.filePath);
+    }
+    return path.join(globalConfig.rootDir, this.filePath);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getIsMonoRepo() {
+    return fs.existsSync(path.join(process.cwd(), 'packages'));
+  }
+
+  ensureResultsDir() {
+    if (!fs.existsSync(this.resultDir)) {
+      fs.mkdirSync(this.resultDir, { recursive: true }, (err) => {
+        if (err) {
+          Logger.error(err.message, { context: LOG_CONTEXT });
+        }
+      });
+    }
+  }
+
+  setTestModule(testLog) {
+    const index = testLog.lastIndexOf('packages/');
+    if (index > -1) {
+      const testFilePath = testLog.substring(index).split('/');
+      const moduleName = testFilePath && testFilePath[1] ? testFilePath[1] : '';
+      if (moduleName && moduleName !== this.moduleName) {
+        this.moduleName = moduleName;
+      }
+    }
+  }
+
+  onRunStart(aggregatedResults) {
+    this.results.startDate = new Date(aggregatedResults.startTime).toLocaleString();
+  }
+
+  log(message) {
+    const readableMessage = `${stripAnsi(message)}${endOfLine}`;
+    this.setTestModule(readableMessage);
+    const moduleName = this.moduleName ? this.moduleName : 'data';
+    if (!this.results.output[moduleName]) {
+      this.results.output[moduleName] = [];
+    }
+    if (!this.unformattedResult[moduleName]) {
+      this.unformattedResult[moduleName] = [];
+    }
+    if (readableMessage.search('\n') !== -1) {
+      this.results.output[moduleName].push(readableMessage.split(/\n/g).forEach((piece) => {
+        this.unformattedResult[moduleName].push(piece);
+      }));
+    }
+    this.results.output[moduleName] = this.unformattedResult[moduleName].filter((obj) => obj);
```

Comment:
Is this filter needed ?
```suggestion
    this.results.output[moduleName] = this.unformattedResult[moduleName]
```